irma-core/state-transitions.js

```diff
@@ -18,67 +18,80 @@ module.exports = {
-    restart:        'Loading',              // Expected payload: undefined
     timeout:        'TimedOut',             // Expected payload: undefined
     abort:          'Aborted',              // Expected payload: undefined
     fail:           'Error'                 // Expected payload: error object
   },
 
-  ShowingIrmaButton: {
-    chooseQR:       'ShowingQRCodeInstead', // Expected payload: undefined
-    appConnected:   'ContinueInIrmaApp',    // Expected payload: undefined
-    abort:          'Aborted',              // Expected payload: undefined
-    fail:           'Error',                // Expected payload: error object
-
-    // We sometimes miss the appConnected transition
-    // on iOS, that's why these transitions are here
-    // too. So we don't 'fail' to the Error state.
-    succeed:        'Success',              // Expected payload: session result (if any)
+  PreparingIrmaButton: {
+    showIrmaButton: 'ShowingIrmaButton',    // Expected payload: {mobile: <app link for launching the IRMA app>}
     cancel:         'Cancelled',            // Expected payload: undefined
     timeout:        'TimedOut',             // Expected payload: undefined
-
-    // State change below is only performed if user agent actually changed.
-    checkUserAgent: 'ShowingQRCode'         // Expected payload: {qr: <payload for in QRs, mobile: <app link for launching the IRMA app>}
+    abort:          'Aborted',              // Expected payload: undefined
+    fail:           'Error',                // Expected payload: error object
   },
 
-  ShowingQRCodeInstead: {
-    appConnected:   'ContinueOn2ndDevice',  // Expected payload: undefined
-    showIrmaButton: 'ShowingIrmaButton',    // Expected payload: undefined
+  ShowingIrmaButton: {
+    chooseQR:       'PreparingQRCode',      // Expected payload: frontend options
+    appConnected:   'ContinueInIrmaApp',    // Expected payload: undefined
+    cancel:         'Cancelled',            // Expected payload: undefined
     timeout:        'TimedOut',             // Expected payload: undefined
     abort:          'Aborted',              // Expected payload: undefined
     fail:           'Error',                // Expected payload: error object
-
-    // State change below is only performed if user agent actually changed.
-    checkUserAgent: 'ShowingQRCode'         // Expected payload: {qr: <payload for in QRs, mobile: <app link for launching the IRMA app>}
+    checkUserAgent: 'CheckingUserAgent',    // Expected payload: undefined
   },
 
   ContinueInIrmaApp: {
-    succeed:        'Success',              // Expected payload: session result (if any)
+    succeed:        'Success',              // Expected payload: undefined
     cancel:         'Cancelled',            // Expected payload: undefined
     timeout:        'TimedOut',             // Expected payload: undefined
     abort:          'Aborted',              // Expected payload: undefined
```

Comment:
This concerns a API change for custom plugins, so we have to make documentation for this to include in the CHANGELOG.